enhancements/olm-managed-metrics.md

```diff
@@ -0,0 +1,146 @@
+---
+title: olm-operators-can-integrate-with-openshift-monitoring
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-11
+last-updated: 2019-12-11
+status: implementable
+---
+
+# olm-operators-can-integrate-with-openshift-monitoring
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The primary purpose of this enhancement is to propose a set of changes that would enable [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) managed operators to define `ServiceMonitors` and `PrometheusRule` objects to interact with the [Prometheus Operator](https://github.com/coreos/prometheus-operator). This feature could then be used on OpenShift clusters to expose metrics to [Telemeter](https://github.com/openshift/telemeter).
+
+For the uninitiated, Red Hat uses metrics collected by Telemeter to:
+
+- Proactively reach out to customers with "problem clusters"
+- Troubleshoot customer issues
+- Understand how individual applications are behaving at a wholistic level
+
+## Motivation
+
+OLM managed operators would significantly benefit from the ability to interact with Prometheus Operators deployed on cluster. By presenting OLM managed operators with the ability to seemlessly integrate with the Prometheus Operator, Cluster Admins would be given a single portal to gauge the health of operators installed on their clusters.
+
+This feature present unique value on OpenShift Clusters, where Telemeter could provide Red Hat with the ability to identify and proactively fix customer issues caused by Red Hat owned operators.
+
+### Goals
+
+- Provide operators with a first-class way of creating `ServiceMonitors` objects
+- Provide operators with a first-class way of creating `PrometheusRules` objects
+- Provide operators with a first-class way of specifying a namespace that they must be deployed to
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+
+## Proposal
+
+### OLM support for new object in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `Namespaces`, `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.ns.yaml
```

Comment:
If the namespace is included as an object in the bundle, it will be in the installplan...which is a namespaced object. Doing it this way will mean that installplans will have to be created in a different namespace than the target CSV.

Reply:
This is a very good point and one that I had not considered. @kevinrizza and @shawn-hurley had requested at different points that the namespace be included in the bundle - would love to hear their thoughts.

Reply:
> Doing it this way will mean that installplans will have to be created in a different namespace than the target CSV.

When you say `in a different namespace` did you mean that the installPlan would need to be a global object?